### ISSUE TITLE
foomatic-db-engine missing database

### ISSUE THREAD

**Issue by Isomorph70** [NixOS/nixpkgs#236450]
### Describe the bug
Error because the foomatic database is not installed.

### Steps To Reproduce
```console
$ nix-shell -p foomatic-db-engine
[nix-shell:~]$ foomatic-searchprinter Phaser
No Foomatic database present!
```

### Expected behavior
Anything but "No Foomatic database present"

### Additional context
Trying to get an old laser printer to work, and want to use foomatic-ppd-to-xml, to remake a new ppd from an old ppd.
But it don't work without a database. 

### Notify maintainers
@Yarny0 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[shs@cynix:~]$  nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.31, NixOS, 23.11 (Tapir), 23.11pre491650.7409480d5c8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by Yarny0** [comment]
Your printer's ppd file might already be contained in the package `foomatic-db-ppds` (or in the unfree alternative `foomatic-db-ppds-withNonfreeDb`).  You can build/fetch that package with `nix-build '<nixpkgs>' -A foomatic-db-ppds -o ./foomatic-db-ppds` (will create a symlink `foomatic-db-ppds` in the CWD), then look around in the directory `foomatic-db-ppds/share/cups/model/foomatic-db-ppds/`.  To use such a ppd file, you might want to use the [NixOS option `hardware.printers.ensurePrinters.*.model`](https://search.nixos.org/options?channel=unstable&show=hardware.printers.ensurePrinters.*.model&from=0&size=50&sort=relevance&type=packages&query=hardware.printers).

If this does not work for you, or you'd like to use the `foomatic-db-engine` package directly for other reasons, you have to obtain at least the `foomatic-db` package and provide its path to the engine via the environment variable `FOOMATICDB`:

```shell
nix-build '<nixpkgs>' -A foomatic-db -o ./foomatic-db  # will create a symlink `foomatic-db` in the CWD
nix-shell -p foomatic-db-engine
FOOMATICDB=$PWD/foomatic-db/share/foomatic foomatic-searchprinter Phaser
```

If you want the content of `foomatic-db-nonfree` included in the database, or if you need to use this environment very often, you can prepare a `shell.nix` file with this content:

<details>
<summary>shell.nix</summary>

 ```nix
{ pkgs ? import <nixpkgs> {} }:

let

  foomatic-db-combined = pkgs.buildEnv {
    name = "foomatic-db-combined";
    pathsToLink = [ "/share/foomatic" ];
    paths = [
      pkgs.foomatic-db
      #pkgs.foomatic-db-nonfree  # add for nonfree data
    ];
  };

in

pkgs.mkShell {
  packages = [ pkgs.foomatic-db-engine ];
  shellHook = "export FOOMATICDB=${foomatic-db-combined}/share/foomatic";
}
```
</details>

With this file in the CWD, you can just start `nix-shell` and have a working foomatic db engine available.

The database for the engine is in separate packages and not hardcoded into the engine package as there are multiple ways combine those database packages to a working database.  Also, it should be possible to use a modified/different database.

**Comment by Isomorph70** [comment]
hmm there seems to be a xml file. /Xerox-Phaser_6125.xml 
in /github.com/OpenPrinting/foomatic-db

**Comment by Isomorph70** [comment]
Yes, I am aware this is a hard case. But I did use it under Linux some years ago. 

Someone figured out that you could use the drivers from Fuji_Xerox-DocuPrint_C525_A_AP-1.0-1.i386.tar,
but the filters in this file is 32bit, so they don't work on 64bit architecture. 

So I was wondering if I could use foomatic-ppd-to-xml on the ppd from this archive, and then use the xml to generate a new ppd file that hopefully use some modern filters. 

I am also new to NixOs, and I have found that working on hard problems force you to dig deep, and then you usually end up
leaning a lot.

**Comment by Isomorph70** [comment]
Hmm, still get an error.

[nix-shell:~/printer]$ echo $FOOMATICDB
/home/shs/printer/foomatic-db/share/foomatic/db/

[nix-shell:~/printer]$ foomatic-ppd-to-xml usr/share/cups/model/FujiXerox/en/FX_DocuPrint_C525_A_AP.ppd 
PPD file usr/share/cups/model/FujiXerox/en/FX_DocuPrint_C525_A_AP.ppd
No Foomatic database present!

**Comment by Isomorph70** [comment]
okey this worked.

`FOOMATICDB=$PWD/foomatic-db/share/foomatic foomatic-ppd-to-xml usr/share/cups/model/FujiXerox/en/FX_DocuPrint_C525_A_AP.ppd`

**Comment by Isomorph70** [comment]
Now I have to figure out, how to generate a ppd from the xml.

**Comment by Yarny0** [comment]
Dear @Isomorph70, did you manage to generate your ppd file?  Can the issue at hand be closed?

**Comment by Isomorph70** [comment]
The original problem is still there. As I see it, the dependencies for the packages are not set up correctly.

**Comment by Yarny0** [comment]
I'm not sure how these dependencies could be defined in a consistent but flexible way.  The problem is that `foomatic-db-engine` is just the engine.  There are at least two possible database "combinations" that might be added as database to be used by the package: `foomatic-db` and `foomatic-db + foomatic-db-nonfree`.  The packages `foomaitc-db-ppds{,-withNonfreeDb}` internally do exactly that: They sets up `foomatic-db-engine` with the proper combinations of database packages.  But someone working on the database might want to use a different (not packaged) version of the database. 

Do you have a specific setup of dependencies in mind that we might implement?
